File: web/src/api/modules/knowledge.ts
import request from '../request'

export interface KnowledgeDocumentOut {
  id: string
  file_id: string
  uploader_id: string
  title: string
  description?: string
  category?: string
  tags?: string[]
  visibility: 'public' | 'teachers_only' | 'private'
  process_status: 'pending' | 'parsing' | 'chunking' | 'embedding' | 'completed' | 'failed'
  chunk_count: number
  summary?: string
  process_error?: string
  processed_at?: string
  created_at: string
  updated_at: string
}

export interface FileOut {
  id: string
  uploader_id: string
  original_name: string
  storage_path: string
  mime_type: string
  file_size: number
  source: string
  created_at: string
}

export interface TeacherAssignedFileOut {
  file: FileOut
  task_id: string
  task_title: string
  task_description?: string
  due_date?: string
  priority?: string
  status?: string
}

export interface CitationItem {
  source_index: number
  document_id: string
  document_title: string
  score: number
}

export interface RAGAnswerOut {
  answer: string
  citations: CitationItem[]
}

export const knowledgeApi = {
  uploadFile(file: File, source = 'upload') {
    const formData = new FormData()
    formData.append('file', file)
    return request.post(`/files/upload?source=${source}`, formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    })
  },

  createDocument(data: {
    file_id: string
    title: string
    description?: string
    category?: string
    tags?: string[]
    visibility?: string
  }) {
    return request.post('/knowledge/documents', data)
  },

  listDocuments(params?: {
    page?: number
    page_size?: number
    category?: string
    visibility?: string
    keyword?: string
  }) {
    return request.get('/knowledge/documents', { params })
  },

  getDocumentDetails(id: string) {
    return request.get(`/knowledge/documents/${id}`)
  },

  updateDocument(id: string, data: {
    title?: string
    description?: string
    category?: string
    tags?: string[]
    visibility?: string
  }) {
    return request.patch(`/knowledge/documents/${id}`, data)
  },

  deleteDocument(id: string) {
    return request.delete(`/knowledge/documents/${id}`)
  },

  listTeacherFiles() {
    return request.get('/knowledge/teacher-files')
  },

  getFileDownloadUrl(fileId: string) {
    return request.get(`/files/${fileId}/download`)
  },

  searchKnowledge(query: string, limit = 5) {
    return request.post(`/knowledge/search?limit=${limit}`, { query })
  },

  knowledgeQA(query: string) {
    return request.post('/knowledge/qa', { query })
  }
}

File: server/app/schemas/knowledge.py
from datetime import datetime
from typing import Optional, List, Any
from uuid import UUID
from pydantic import BaseModel, Field

class FileOut(BaseModel):
    id: UUID
    uploader_id: UUID
    original_name: str
    storage_path: str
    mime_type: str
    file_size: int
    source: str
    created_at: datetime

    class Config:
        from_attributes = True

class KnowledgeDocumentBase(BaseModel):
    title: str = Field(..., max_length=255, description="文档标题")
    description: Optional[str] = Field(None, description="文档详情/描述")
    category: Optional[str] = Field("other", description="分类")
    tags: Optional[List[str]] = Field(None, description="标签列表")
    visibility: str = Field("public", description="可见性: public, teachers_only, private")

class KnowledgeDocumentCreate(KnowledgeDocumentBase):
    file_id: UUID = Field(..., description="关联上传的文件ID")

class KnowledgeDocumentUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=255, description="Document title")
    description: Optional[str] = Field(None, description="Document description")
    category: Optional[str] = Field(None, description="Folder/category")
    tags: Optional[List[str]] = Field(None, description="Tag list")
    visibility: Optional[str] = Field(None, description="public, teachers_only, private")

class KnowledgeDocumentOut(KnowledgeDocumentBase):
    id: UUID
    file_id: UUID
    uploader_id: UUID
    process_status: str
    chunk_count: int
    summary: Optional[str] = None
    process_error: Optional[str] = None
    processed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class RAGQueryReq(BaseModel):
    query: str = Field(..., min_length=1, description="提问/检索关键字")

class CitationItem(BaseModel):
    source_index: int
    document_id: str
    document_title: str
    score: float

class RAGAnswerOut(BaseModel):
    answer: str
    citations: List[CitationItem] = []

class TeacherAssignedFileOut(BaseModel):
    file: FileOut
    task_id: UUID
    task_title: str
    task_description: Optional[str] = None
    due_date: Optional[datetime] = None
    priority: Optional[str] = None
    status: Optional[str] = None
